### PR TITLE
Add Serialization and Deserialization support to MerkleRootCalculator

### PR DESCRIPTION
**Pull request by Salka1988**
Adds serialization and deserialization support for the MerkleRootCalculator struct. 

Additional work: 
- getter for MerkleRootCalculator stack
- constructor new_with_stack

### CODE REVIEW

**Review by bvrooman — COMMENTED** [review]
I suggest to put serialization/deserialization behind a `serde` feature flag, similar to something like https://github.com/FuelLabs/fuel-core/blob/master/crates/types/Cargo.toml#L31

File: fuel-merkle/src/binary/node.rs [hunk 1]
```diff
@@ -10,8 +10,12 @@ use crate::{
 };
 
 use core::fmt::Debug;
-
+// use serde::{
+//     Deserialize,
+//     Serialize,
+// };
```
Comment (xgreenx) [inline fuel-merkle/src/binary/node.rs:16]:
```suggestion
```

File: fuel-merkle/src/binary/root_calculator.rs [hunk 2]
```diff
@@ -9,7 +9,11 @@ use crate::{
 use crate::alloc::borrow::ToOwned;
 use alloc::vec::Vec;
 
-#[derive(Default, Debug, Clone)]
+#[cfg(test)]
+use serde_json as _;
```
Comment (xgreenx) [inline fuel-merkle/src/binary/root_calculator.rs:13]: You can move it to `mod test` section

File: fuel-merkle/src/common/position.rs [hunk 3]
```diff
@@ -8,6 +8,10 @@ use crate::common::{
     PositionPath,
 };
 use core::convert::Infallible;
+// use serde::{
+//     Deserialize,
+//     Serialize,
+// };
```
Comment (xgreenx) [inline fuel-merkle/src/common/position.rs:14]:
```suggestion
```

File: fuel-merkle/Cargo.toml [hunk 4]
```diff
@@ -24,12 +25,16 @@ datatest-stable = "0.1"
...
 hex = "0.4"
 rand = "0.8"
+serde_json = "1.0"
 serde_yaml = "0.9"
 
 [features]
 default = ["std"]
 std = ["digest/default", "hex/default", "sha2/default"]
 test-helpers = []
+serde = [
```
Comment (xgreenx) [inline fuel-merkle/Cargo.toml:35]: `fuel-vm` and `fuel-tx` crates has `server` feature too. Could you add `fuel-merkle/serde` there too please?=)
image

**Review by xgreenx — APPROVED** [review]
Thank you! <3

**Pull request merged**